### PR TITLE
fix: remove host user's home directory mount from the KIB wrapper container 

### PR DESCRIPTION
**Pull request by supershal**
**What problem does this PR solve?**:
- remove mounting host user's home directory on container to reduce security risk
- mount credentials and ssh file mounts as readonly

**Which issue(s) does this PR fix?**:

* https://d2iq.atlassian.net/browse/D2IQ-93781

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cmd/konvoy-image-wrapper/cmd/wrapper.go [hunk 1]
```diff
@@ -150,6 +152,11 @@ func (r *Runner) setAWSEnv() error {
 		}
 	}
 
+	_, found := os.LookupEnv(envAWSCredentialsFile)
+	if !found {
+		// fall-back to default location for aws credentials file
+		os.Setenv(envAWSCredentialsFile, filepath.Join(r.usr.HomeDir, ".aws", "credentials"))
```
Comment (supershal) [inline cmd/konvoy-image-wrapper/cmd/wrapper.go:158]: When user's homedir mount was removed, the aws creds from `credentials` stopped working. So defaulting `AWS_SHARED_CREDENTIALS_FILE` to ~/.aws/credentials as per the variable's documentation given at https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html

Comment (faiq) [reply cmd/konvoy-image-wrapper/cmd/wrapper.go:158]: Do we have to review this for SSH agent socket or other cloud providers?

Comment (supershal) [reply cmd/konvoy-image-wrapper/cmd/wrapper.go:158]: I can make ssh agent socket readonly and test it out. I am not sure if packer or KIB writes to this socket.
The wrapper mounts GCP credentials file to the container. Mounting files pointed by credentials env variable was moved to a common function [mountFileEnv](https://github.com/mesosphere/konvoy-image-builder/pull/589/files#diff-d8016cb551cfda4bbd3bc9124263f9003367a4cf5a6d144b96c928faec12f3aaR217) in last refactor. so above fix will apply to all the cloud provider's credentials.

Comment (faiq) [reply cmd/konvoy-image-wrapper/cmd/wrapper.go:158]: Sounds good. Let me know how it goes.

Comment (supershal) [reply cmd/konvoy-image-wrapper/cmd/wrapper.go:158]: mounted `SSH_AUTH_SOCK` as readonly: https://github.com/mesosphere/konvoy-image-builder/pull/589/commits/c4368fbfd316b22223e920f3c92f0da0b7d23ed6

File: cmd/konvoy-image-wrapper/cmd/wrapper.go [hunk 2]
```diff
@@ -308,7 +315,14 @@ func (r *Runner) dockerRun(args []string) error {
...
-		cmd.Args = append(cmd.Args, "--mount", "type="+v.kind+",source="+v.source+",target="+v.target)
+		bindOptions := ""
+		if len(v.bindOptions) > 0 {
+			bindOptions = fmt.Sprintf(",%s", strings.Join(v.bindOptions, ","))
+		}
+		cmd.Args = append(
+			cmd.Args,
+			"--mount",
+			"type="+v.kind+",source="+v.source+",target="+v.target+bindOptions)
 	}
```
Comment (faiq) [inline cmd/konvoy-image-wrapper/cmd/wrapper.go:326]: nit: refactoring this to a function would be nice for code readability

Comment (supershal) [reply cmd/konvoy-image-wrapper/cmd/wrapper.go:326]: I am not too sure about moving this to function. This function will be only called from this loop. If we have multiple callers for the same functionality we can think of factoring out it a function.

**Review by faiq — APPROVED** [review]

**Pull request merged**